frontend/src/components/MonacoLanguageServerAugmentation.ts

```diff
@@ -19,23 +20,82 @@ monaco.languages.register({
     aliases: ['p4', 'P4']
 })
 
-export default (editor: monaco.editor.IStandaloneCodeEditor) : monaco.editor.IStandaloneCodeEditor=> editor
-
-export const LSAugmentation =  (editor: monaco.editor.IStandaloneCodeEditor) : monaco.editor.IStandaloneCodeEditor=> {
-    // install Monaco language client services
-    // @ts-ignore
-    MonacoServices.install(editor,{rootUri: "file://tmp"});
-    const webSocket = createWebSocket('/environment/p4basic/languageserver/p4');
-    // listen when the web socket is opened
-    listen({
-        webSocket,
-        onConnection: connection => {
-            // create and start the language client
-            const languageClient = createLanguageClient(connection);
-            const disposable = languageClient.start();
-            connection.onClose(() => disposable.dispose());
-        }
-    });
+monaco.languages.register({
+    id: 'python',
+    extensions: ['.py'],
+    aliases: ['Python', 'py', 'PY', 'python']
+})
+
+// register Monaco languages
+monaco.languages.register({
+    id: 'json',
+    extensions: ['.json', '.bowerrc', '.jshintrc', '.jscsrc', '.eslintrc', '.babelrc'],
+    aliases: ['JSON', 'json'],
+    mimetypes: ['application/json'],
+});
+
+//export default (editor: monaco.editor.IStandaloneCodeEditor, path: string) : monaco.editor.IStandaloneCodeEditor=> editor
```

Comment:
Sure! This particular one came from you, and allowed quickly switching between language server integration and the standalone version of the editor, but if we need that again, it can easily be dragged from the history. Yes! Code is of course tested and works. Due to some shortcomings in p4 language server editing p4 code lags a bit, which does not happen, e.g. for Python with py-ls. Anyway, this could also be influenced by disabling p4 language server support in the introduced MonacoLanguageSelector.

Reply:
You can try the new implementation. My local testing shows that it is pretty fast.